### ISSUE TITLE
Can not test with useRequest for fail case when mocking api call

### ISSUE THREAD

**Issue by tonynguyenit18** [alibaba/hooks#1001]
I have an api e.g

```ts
// api.ts
export const getUser = () =>{
  return axios.post("domain/get-user")
}
```

Then I have a custom hook
```ts
export const useGetUser = () => {
     const  {loading, error, run}  = useRequest(
      async () => {
            const response = await getUser()
            console.log("response", response)
            return response
      }
     ),
    {
      onSuccess: payload => {
          console.log("payload", payload)
      },
      onError: err => {
        console.log("err", err)
      }
    }
  )
}
```

OR

```ts
export const useGetUser = () =>{
  const {loading, error, run} = useRequest(
      async () => {
             try{
               const response = await getUser()
               console.log("response", response)
               return response
             }catch(err){
                ....
               throw err
           }
        })
}
```

I am doing a test for my customer hook using react-testing-library and mock the api call
```ts
import * as APIS from. "./api"
describe("useGetUser", () => {
   it("fail test case", () => {
    const mockError: any = {
      response: {
        data: {
          msg: "Error Message"
                },
        status: 401
      }
    }
    jest.spyOn(APIS,"getUser").mockRejectedValue(mockError)
    const { result, waitForNextUpdate} = renderHook(() =>useGetUser())
   ....
  }   
 )
}
```
I will get error
`thrown: "useRequest has caught the exception, if you need to handle the exception yourself, you can set options.throwOnError to true."`

As long as I mock throwing error from first param of `useRequest` I'll get this issue

I guess it from 
From https://github.com/alibaba/hooks/blob/master/packages/use-request/src/useAsync.ts#L174

**Comment by brickspert** [comment]
Yes, this is a bug for useRequest, we will fixed this in next major version.

**Comment by tonynguyenit18** [comment]
> Yes, this is a bug for useRequest, we will fixed this in next major version.

Thanks @brickspert for quick reply. 
Is anyone on this task? I can take it.

**Comment by brickspert** [comment]
Hello：

这个问题我们已经在 v3.0 版本彻底修复，目前 v3.0 版本已经发布 alpha 版本。

欢迎试用。

文档：https://ahooks-next.surge.sh/
源码：https://github.com/alibaba/hooks/tree/next

安装最新分支
```
$ npm install --save ahooks@next
# or
$ yarn add ahooks@next
```

We have fixed this problem in the v3.0 version, and the alpha has been released. 

Welcome to try.

Doc：https://ahooks-next.surge.sh/
Source Code：https://github.com/alibaba/hooks/tree/next

Install v3.0

```
$ npm install --save ahooks@next
# or
$ yarn add ahooks@next
```
